compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -737,7 +737,7 @@ impl<'p, 'tcx: 'p> RustcPatCtxt<'p, 'tcx> {
                 // this). We show this to the user as `usize::MAX..` which is slightly incorrect but
                 // probably clear enough.
                 let c = ty.numeric_max_val(cx.tcx).unwrap();
-                let value = mir::Const::from_ty_const(c, cx.tcx);
+                let value = mir::Const::from_ty_const(c, self.tcx.types.usize, cx.tcx);
```

Comment:
This is incorrect, we can also get `isize` here.
```suggestion
                let value = mir::Const::from_ty_const(c, ty, cx.tcx);
```

Reply:
What's weird is that there are tests for the `isize` case in `tests/ui/pattern/usefulness/integer-ranges/pointer-sized-int.rs`. Do you use that type argument atm?

Reply:
shit thanks